src/Http/Http/src/Internal/DefaultConnectionInfo.cs

```diff
@@ -44,6 +47,9 @@ public void Uninitialize()
         private ITlsConnectionFeature TlsConnectionFeature=>
             _features.Fetch(ref _features.Cache.TlsConnection, _newTlsConnectionFeature)!;
 
+        private IConnectionLifetimeNotificationFeature ConnectionLifetime =>
+            _features.Fetch(ref _features.Cache.ConnectionLifetime, _newConnectionLifetime)!;
```

Comment:
Can't this be null?

Reply:
This is one of my questions that if a server does not have `IConnectionLifetimeNotificationFeature` should there be a default implementation of `IConnectionLifetimeNotificationFeature` which at least trigger its cancellation token on `RequestClose`?

Reply:
The default should look like the default Abort feature, plus maybe adding the close header:
https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http/src/Features/HttpRequestLifetimeFeature.cs